### PR TITLE
Event grid test resources

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Set up test resource infrastructure for Track 2 SDK.
Add a tests.yml file to set up live test pipeline. It doesn't seem as though it is possible to test this from the PR, as the yml file needs to already be merged in order to trigger live test pipeline. I believe this would trigger the Track 1 SDK tests as well so it would be nice to verify that those will work. There are no Track 2 tests yet.

### CODE REVIEW

File: sdk/eventgrid/Azure.Messaging.EventGrid/src/autorest.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Generated code configuration
+
+Run `dotnet msbuild /t:GenerateCode` to generate code.
+
+``` yaml
+input-file:
+    -  https://github.com/ellismg/azure-rest-api-specs/blob/e93e9529ffa1a3ace9b260070de4bfdf57be17e4/specification/eventgrid/data-plane/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json
```
Comment (JoshLove-msft) [inline sdk/eventgrid/Azure.Messaging.EventGrid/src/autorest.md:7]: This is the current swagger prototype that includes cloud events.

File: sdk/eventgrid/Azure.Messaging.EventGrid/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Release History
+
+## 1.0.2-preview.14 (2020-06-02)
```
Comment (pakrym) [inline sdk/eventgrid/Azure.Messaging.EventGrid/CHANGELOG.md:3]: Clean this up

File: sdk/eventgrid/Azure.Messaging.EventGrid/src/Azure.Messaging.EventGrid.csproj [hunk 3]
```diff
@@ -0,0 +1,34 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
```
Comment (pakrym) [inline sdk/eventgrid/Azure.Messaging.EventGrid/src/Azure.Messaging.EventGrid.csproj:2]: Clean this up.

File: sdk/eventgrid/Azure.Messaging.EventGrid/test-resources.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        "eventGridDomainName": "[concat(parameters('baseName'), 'domain')]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.EventGrid/topics",
+            "apiVersion": "[variables('apiVersion')]",
+            "name": "[variables('eventGridTopicName')]",
+            "location": "[resourceGroup().location]"
+        },
+		{
```
Comment (pakrym) [inline sdk/eventgrid/Azure.Messaging.EventGrid/test-resources.json:21]: nit: tabs vs spaces

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/EventGridRecordedTestSanitizer.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        // These constants should eventually go in our product code, as they will be needed when creating
+        // the AzureKeyCredential.
+        public const string SasKeyName = "aeg-sas-key";
+        public const string SasTokenName = "aeg-sas-token";
+
+        public override void SanitizeHeaders(IDictionary<string, string[]> headers)
+        {
+            base.SanitizeHeaders(headers);
+
+            if (headers.TryGetValue(SasKeyName, out var sasKey))
```
Comment (pakrym) [inline sdk/eventgrid/Azure.Messaging.EventGrid/tests/EventGridRecordedTestSanitizer.cs:21]: Should we make https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core.TestFramework/src/RecordedTestSanitizer.cs#L20 a list and public?

**Review by pakrym — APPROVED** [review]

**Comment by JoshLove-msft** [comment]
@weshaggard will this PR cause existing Track 1 Event Grid tests to start running nightly? There doesn't appear to be an existing test pipeline for Event Grid, but they do have live tests.
/cc @ahamad-MS

File: sdk/eventgrid/Azure.Messaging.EventGrid/src/Azure.Messaging.EventGrid.csproj (hunk 3 above)
Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/src/Azure.Messaging.EventGrid.csproj:2]: I'd rather leave this as something that will be done as part of summer project.

**Comment by weshaggard** [comment]
>@weshaggard Wes Haggard FTE will this PR cause existing Track 1 Event Grid tests to start running nightly? 

Yes I do believe that might start happening as we build all projects in the service directory (see https://github.com/Azure/azure-sdk-for-net/blob/master/eng/service.proj#L10). Once the test pipeline is setup you can always test it and if needed we can disable the other test projects.

**Pull request merged**